types/oscal/oscal.go

```diff
@@ -183,30 +148,34 @@ func New(r io.Reader) (*OSCAL, error) {
 				return &OSCAL{Profile: &profile}, nil
 			}
 		}
-
-		return nil, errors.New("Malformed OSCAL")
 	}
 
 	return nil, errors.New("Malformed OSCAL. Must be XML or JSON")
 }
 
-// RawXML ...
-func (o *OSCAL) RawXML(prettify bool) ([]byte, error) {
+// XML writes the OSCAL object as XML to the given writer
+func (o *OSCAL) XML(w io.Writer, prettify bool) error {
+	e := xml.NewEncoder(w)
 	if prettify {
-		return xml.MarshalIndent(o, "", "  ")
+		e.Indent("", "  ")
+		return e.Encode(o)
```

Comment:
This statement might not be needed as it’s already there at the end of the method (line 164)